actioncable/lib/action_cable/connection/base.rb

```diff
@@ -170,7 +170,8 @@ def handle_open
           message_buffer.process!
           server.add_connection(self)
         rescue ActionCable::Connection::Authorization::UnauthorizedError
-          respond_to_invalid_request
+          transmit type: ActionCable::INTERNAL[:message_types][:unauthorized]
```

Comment:
Is there a guarantee that client would receive this message before the connection is closed? I think, it depends on the client implementation (note that there could be non-js clients too).

I was thinking about this feature, and my idea was to use [close codes](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent) (which are not supported by Action Cable currently)

Reply:
It works consistently using the included JS client. The server will send the text frame before it sends the close frame. Not sure why clients would not receive them in that order?

Reply:
> Not sure why clients would not receive them in that order?

Agree. Frames are written sequentially and synchronously.